### PR TITLE
Add version guard in Istiod telemetry filter config for 1.9+ proxies

### PR DESCRIPTION
**Pull request by carolynhu**
Please provide a description for what this PR is for.

https://github.com/istio/istio/issues/28410
Only triggering Istiod telemetry filter config for proxy version >= 1.9

Tested locally:
- Installed istio 1.8.2
- Build 1.9 istioctl, and install istio 1.9
- Check the istio-ingressgateway istio-proxy container image is `gcr.io/istio-testing/proxyv2:1.9-dev` and checked the config_dump, there is no duplicate filters being inserted.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -239,6 +240,10 @@ func (configgen *ConfigGeneratorImpl) BuildListeners(node *model.Proxy,
 	push *model.PushContext) []*listener.Listener {
 	builder := NewListenerBuilder(node, push)
 
+	if util.IsIstioVersionGE19(node) {
+		configgen.Plugins = append(configgen.Plugins, metadataexchange.NewPlugin())
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go:244]: Why does this fix the problem?  don't we still install the 1.9 mx Envoyfilter?

Comment (carolynhu) [reply pilot/pkg/networking/core/v1alpha3/listener.go:244]: we need to remove the yaml section related to that part ... i'm working on adding the missing cluster filter for tcp, once I done that I can remove those 1.9 mx envoyfilter.

Comment (bianpengyuan) [reply pilot/pkg/networking/core/v1alpha3/listener.go:244]: We probably should target 1.10, considering it is pretty late in the release cycle.

File: pilot/pkg/networking/networking.go [hunk 2]
```diff
@@ -118,6 +120,9 @@ type MutableObjects struct {
 
 	// FilterChains is the set of filter chains that will be attached to Listener.
 	FilterChains []FilterChain
+
+	// Cluster is the set of clusters that will be attached to Cluster.
```
Comment (howardjohn) [inline pilot/pkg/networking/networking.go:124]: Sorry, a bit confused. How is a cluster attached to a cluster?

File: pilot/pkg/networking/networking.go [hunk 3]
```diff
@@ -15,11 +15,13 @@
 package networking
 
 import (
+	cluster1 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
 	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
 	http_conn "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
 	thrift_proxy "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/thrift_proxy/v3"
 	tls "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	_ "google.golang.org/genproto/googleapis/bigtable/admin/cluster/v1"
```
Comment (howardjohn) [inline pilot/pkg/networking/networking.go:24]: what is this for?

File: pilot/pkg/networking/plugin/authn/authentication.go [hunk 4]
```diff
@@ -34,6 +34,10 @@ var (
 // Plugin implements Istio mTLS auth
 type Plugin struct{}
 
+func (p Plugin) OnOutboundCluster(mutable *networking.MutableObjects) error {
+	panic("implement me")
```
Comment (howardjohn) [inline pilot/pkg/networking/plugin/authn/authentication.go:38]: We should not panic

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: pilot/pkg/networking/networking.go (hunk 2 above)
Comment (carolynhu) [reply pilot/pkg/networking/networking.go:124]: oops, sorry, should be filters attached to cluster..

**Pull request closed without merging**